keen/explorer
# Fix group-by fields and Timeframe type toggle (PLUS: lodash upgrade)

## What does this PR do? How does it affect users?

It fixes two bugs in Explorer:

1) When adding a second group by option, you could not remove that second group by field after adding it. This was due to how updates are merged into the model, a change that was introduced in #98.
2) When trying to toggle between absolute and relative timeframes, it would not work. You'd get "stuck" on an absolute timeframe after selecting it. This was due to how updates are merged into the model, a change that was introduced in #98.

There are two new tests added to cover these two scenarios.

#### PLUS: Upgrade to lodash 4.0.0
This PR also upgrades to latest lodash because we needed the `_.mergeWith` function to fix these bugs. :palm_tree: 

## How should this be tested?

1) Try using the group by fields in the query builder. Ensure they work as expected.
2) Try toggling between absolute and relative timeframes. Ensure they work as expected.

Step through the code line by line. Things to keep in mind as you review:
 - Are there any edge cases not covered by this code?
 - Does this code follow conventions (naming, formatting, modularization, etc) where applicable?

Fetch the branch and/or deploy to staging to test the following:
